modules/user/user.controller.js

```diff
@@ -1,66 +1,109 @@
 const express = require("express");
 var router = express.Router(); // - /v1/user/
-
-const filePath = "./database/users.json";
 const jsonParser = express.json();
-const fs = require("fs");
-
-//получение всех пользователей
-router.get("/", function (req, res) {
-    fs.readFile(filePath, function (err, content) {
-        if (err) {
-            res.statusCode = 500;
-            res.end("Server error");
-            return;
-        }
+const { isEmpty, getUser } = require("../../utils.js");
+const userService = require("./user.service");
 
-        const users = JSON.parse(content);
+router.get("/", async function (req, res) {
+    try {
+        let users = await userService.readAllUsers();
         res.send(users);
-    });
+    } catch (err) {
+        res.statusCode = 500;
+        res.end("Server error");
+    }
 });
 
-//получение конкретного пользователя
-router.get("/:id", function (req, res) {
-    const id = req.params.id;
-    let userById = null;
+router.get("/:id", async function (req, res) {
+    try {
+        const id = req.params.id;
+
+        let user = await userService.readOneUser(id);
 
-    fs.readFile(filePath, function (err, content) {
-        if (err) {
+        res.send(user);
+    } catch (err) {
+        if (err.message == 404) {
+            res.statusCode = 404;
+            res.end("User not found");
+        } else {
             res.statusCode = 500;
             res.end("Server error");
-            return;
         }
+    }
+});
 
-        const users = JSON.parse(content);
-        for (let user of users) {
-            if (user.id == id) {
-                userById = user;
-                break;
-            }
+router.post("/", jsonParser, async function (req, res) {
+    try {
+        if (isEmpty(req.body)) {
+            throw new Error(400);
         }
 
-        if (userById) {
-            res.send(userById);
+        let user = getUser(req);
+
+        let newUser = await userService.createUser(user);
+
+        res.send(newUser);
+    } catch (err) {
+        if (err.message == 400) {
+            res.statusCode = 400;
+            res.end("Bad Request");
         } else {
-            res.statusCode = 404;
-            res.end("User not found");
+            res.statusCode = 500;
+            res.end("Server error");
         }
-    });
+    }
 });
 
-//создание пользователя
-router.post("/", function (req, res) {
-    res.send("post");
-});
+router.put("/:id", jsonParser, async function (req, res) {
+    try {
+        if (isEmpty(req.body)) {
+            throw new Error(400);
+        }
 
-//обновление пользователя
-router.put("/", function (req, res) {
-    res.send("put");
+        let user = getUser(req);
+
+        let newUser = await userService.updateUser(user);
+
+        res.send(newUser);
+    } catch (err) {
+        switch (err.message) {
```

Comment:
this code with error handling duplicated few time, so it's better to move it to some error handling function